pkg/rbac/rbac.go

```diff
@@ -0,0 +1,287 @@
+	resourceACLs, err := GetResourceAccess(userKClient, MetricsACLConfig.gr, clusters, "")
+	if err != nil {
+		return nil, err
+	}
+
+	klog.V(2).Infof(" resource access results: %v", resourceACLs)
+
+	// from the list of all ACLS on ManagedCluster, filter out the "metrics" specific acls and grab the namespaces
+	metricsAccessResults := make(map[string][]string)
+
+	for clustername, clusteracls := range resourceACLs {
+		klog.V(2).Infof("clustername [%s] acls[%s]\n", clustername, clusteracls)
+
+		// list of namespaces for the cluster
+		nsWithMetricsAccess := []string{}
+
+		for _, acl := range clusteracls {
+			// filter verbs that start with metrics/ and grab the namespace
+			// if verb is set to *, set namespace to * to indicate access to all namespaces
+			if strings.HasPrefix(acl, MetricsACLConfig.verb) {
+				nsWithMetricsAccess = addUniqueItems(
+					nsWithMetricsAccess, strings.TrimPrefix(acl, MetricsACLConfig.verb))
+			} else if acl == "*" {
+				nsWithMetricsAccess = addUniqueItems(nsWithMetricsAccess, "*")
+			}
+		}
+
+		klog.V(2).Infof("clustername [%s], Namespaces with metrics access[%s]\n", clustername, nsWithMetricsAccess)
+		metricsAccessResults[clustername] = nsWithMetricsAccess
+	}
+
+	klog.V(2).Infof(" metricsAccessResults is %v", metricsAccessResults)
+
+	return metricsAccessResults, nil
+}
+
+// returns all Acls for a given resource type
+// if resource names is non-empty, acls are filtered for only the given list of resourcenames ,
+// else if resource names is empty,acls for resources of the given type that user has access to are returned
+// namespace is used for namespace-scoped resources, for cluster-scoped resources is can be left empty
+// for a given cluster, if no  acls are configured, an empty list is returned for it in the results
+
+func GetResourceAccess(
+	kclient kubernetes.Interface, gr schema.GroupResource, resourcenames []string, namespace string,
+) (map[string][]string, error) {
+	klog.V(2).Infof(
+		"GetResourceAccess for GroupResource: %s, resourcenames: %v, namespace: %s", gr, resourcenames, namespace)
+
+	// make a SelfSubjectRulesReview to get all resource rules configured.
+	resourceRules, err := MakeSubjectAccessRulesReviewForUser(kclient, namespace)
+	if err != nil {
+		return nil, err
+	}
+
+	resourceAccessResults := make(map[string][]string)
+	// search through all the resource rules
+	for _, rule := range resourceRules {
+		// each resource rule contains { []ApiGroup, []Resources, []ResourceNames, []Verbs}
+		// e.g: {[metrics/nsred1 metrics/nsred2] [cluster.open-cluster-management.io]
+		// [managedclusters] [devcluster1 devcluster2]}
+		// filter the rules by the given ApiGroup(or *) and Resource(or *))
+		ruleMatchesAPIGroup := (slices.Contains(rule.APIGroups, gr.Group) || slices.Contains(rule.APIGroups, "*"))
+		ruleMatchesResource := (slices.Contains(rule.Resources, gr.Resource) || slices.Contains(rule.Resources, "*"))
+		if ruleMatchesAPIGroup && ruleMatchesResource {
```

Comment:
Optional: To reduce the indentation and improve readability.
```suggestion
		if !(ruleMatchesAPIGroup && ruleMatchesResource) {
			continue
		}
```